[framework] Add Look Up to selection controls for iOS

LouiseHsu opened the pull request:
This PR adds framework support for the Look Up feature in iOS. 

https://github.com/flutter/flutter/assets/36148254/d301df79-4e23-454f-8742-2f8e39c2960c

The corresponding merged engine PR can be found [here](https://github.com/flutter/engine/pull/43308).
This PR addresses https://github.com/flutter/flutter/issues/82907 
More details are available in this [design doc.](flutter.dev/go/add-missing-features-to-selection-controls)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

LouiseHsu committed:
initial pr commit

LouiseHsu committed:
Merge branch 'master' into add-lookup-to-selection-controls

justinmc reviewed (commented):
I just took a quick look at this since it's in draft mode, but the approach for adding the new button looks good.

It looks like if someone were to use the iOS-style context menu on a different platform, then they would still see the Look Up button, but tapping it wouldn't do anything.  I guess that's probably the right behavior?  It would be really weird to do that for the iOS menu, though for other platforms it might make more sense (someone using the Windows/Linux AdaptiveDesktopTextSelectionToolbar on Mac for example).  The other option is to conditionally show Look Up only on iOS.

**Edit**: Also, the link in the description to the "Design doc" seems to not be correct FYI.

polina-c committed:
Enable not GCed leak tracking. (#130159)

engine-flutter-autoroll committed:
Roll Flutter Engine from 875d87e02276 to d00590fde18c (5 revisions) (#130369)

https://github.com/flutter/engine/compare/875d87e02276...d00590fde18c

2023-07-11 jason-simmons@users.noreply.github.com Release the mutex in the CanCreateConcurrentMessageLoop test before waking the latch (flutter/engine#43574)
2023-07-11 skia-flutter-autoroll@skia.org Roll Skia from 2998197ce4ca to c769464a8b79 (2 revisions) (flutter/engine#43577)
2023-07-11 zanderso@users.noreply.github.com Run linux_unopt on a machine with more cores (flutter/engine#43473)
2023-07-11 skia-flutter-autoroll@skia.org Roll Skia from d2051446ee6b to 2998197ce4ca (1 revision) (flutter/engine#43572)
2023-07-11 skia-flutter-autoroll@skia.org Roll Dart SDK from 8b1c04086525 to 551915bf6ebe (1 revision) (flutter/engine#43571)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from d00590fde18c to e109a3c0c347 (1 revision) (#130371)

https://github.com/flutter/engine/compare/d00590fde18c...e109a3c0c347

2023-07-12 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from qVHNy9nU6cQKtR_IB... to LV7zVdnjfEPukVG8L... (flutter/engine#43580)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from qVHNy9nU6cQK to LV7zVdnjfEPu

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from e109a3c0c347 to 1b44d7ee1a60 (3 revisions) (#130380)

https://github.com/flutter/engine/compare/e109a3c0c347...1b44d7ee1a60

2023-07-12 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from EnRnFf_eyS6SGqpHi... to zTuZjXzI3bqscWmeo... (flutter/engine#43583)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from c769464a8b79 to d0ecba6d753c (2 revisions) (flutter/engine#43582)
2023-07-12 skia-flutter-autoroll@skia.org Roll Dart SDK from 551915bf6ebe to 51d6128b190e (1 revision) (flutter/engine#43581)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from EnRnFf_eyS6S to zTuZjXzI3bqs

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 1b44d7ee1a60 to 7de68c62742d (4 revisions) (#130384)

https://github.com/flutter/engine/compare/1b44d7ee1a60...7de68c62742d

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 7b0d85cff28a to 33cfa4fc2aeb (1 revision) (flutter/engine#43586)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from d0ecba6d753c to 7b0d85cff28a (1 revision) (flutter/engine#43585)
2023-07-12 chinmaygarde@google.com [Impeller] Remove un-actionable info log about validations. (flutter/engine#43575)
2023-07-12 chinmaygarde@google.com [Impeller] Create a resource manager that batch deallocates resources on a dedicated thread. (flutter/engine#43579)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

huycozy committed:
Make new issue template for 1P package (#130065)

Closes https://github.com/flutter/flutter/issues/117750.

### Demo
<details>
<summary>Preview screenshot</summary>

![screenshot](https://github.com/flutter/flutter/assets/104349824/40968935-b898-4303-b3ee-08abed94f6eb)
</details>

Demo issue template: [Report a bug in Flutter's first party package](https://github.com/huycozy/flutter/issues/new?assignees=&labels=&projects=&template=9_first_party_packages.yml)
Demo a filed issue: https://github.com/huycozy/flutter/issues/2

engine-flutter-autoroll committed:
Roll Flutter Engine from 7de68c62742d to 3482e05776a7 (2 revisions) (#130390)

https://github.com/flutter/engine/compare/7de68c62742d...3482e05776a7

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from af2a829258a0 to 0ac60b2cecfe (2 revisions) (flutter/engine#43592)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 33cfa4fc2aeb to af2a829258a0 (1 revision) (flutter/engine#43591)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 3482e05776a7 to 29c5c41eeb19 (1 revision) (#130393)

https://github.com/flutter/engine/compare/3482e05776a7...29c5c41eeb19

2023-07-12 skia-flutter-autoroll@skia.org Roll Dart SDK from 51d6128b190e to b95f6531c726 (2 revisions) (flutter/engine#43594)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 29c5c41eeb19 to 73093fdd77c3 (2 revisions) (#130398)

https://github.com/flutter/engine/compare/29c5c41eeb19...73093fdd77c3

2023-07-12 flar@google.com header file cleanup focusing on removing unnecessary SkPicture includes (flutter/engine#43589)
2023-07-12 bdero@google.com [Impeller] Document ColorSourceContents (flutter/engine#43590)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 73093fdd77c3 to 3dbe7dbeb5d4 (1 revision) (#130399)

https://github.com/flutter/engine/compare/73093fdd77c3...3dbe7dbeb5d4

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 0ac60b2cecfe to 0fb595ccc6a7 (1 revision) (flutter/engine#43596)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 3dbe7dbeb5d4 to c7317a58466e (1 revision) (#130402)

https://github.com/flutter/engine/compare/3dbe7dbeb5d4...c7317a58466e

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 0fb595ccc6a7 to 6a8be5964fbe (1 revision) (flutter/engine#43597)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from c7317a58466e to d68ea304eeda (1 revision) (#130410)

https://github.com/flutter/engine/compare/c7317a58466e...d68ea304eeda

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 6a8be5964fbe to 9038aec6a23a (1 revision) (flutter/engine#43598)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from d68ea304eeda to 5c887028810d (2 revisions) (#130413)

https://github.com/flutter/engine/compare/d68ea304eeda...5c887028810d

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 9038aec6a23a to ac4c113c071d (1 revision) (flutter/engine#43600)
2023-07-12 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from LV7zVdnjfEPukVG8L... to 0fvk838jTDNQ_l43k... (flutter/engine#43599)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from LV7zVdnjfEPu to 0fvk838jTDNQ

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

TahaTesser committed:
Update `RadioListTile` tests format for M2/M3 (#130391)

Update tests format 

context: https://github.com/flutter/flutter/pull/129718#issuecomment-1615124857

engine-flutter-autoroll committed:
Roll Packages from 188a8468f557 to 250871431bed (14 revisions) (#130418)

https://github.com/flutter/packages/compare/188a8468f557...250871431bed

2023-07-12 39314023+wer-mathurin@users.noreply.github.com ADD appBarBreakpoint (flutter/packages#4434)
2023-07-12 engine-flutter-autoroll@skia.org Roll Flutter from 65ff3cb9b45c to 3ec96a8a4a5d (5 revisions) (flutter/packages#4415)
2023-07-12 tarrinneal@gmail.com [image_picker] Roll dependancies to avoid error (flutter/packages#4431)
2023-07-12 49699333+dependabot[bot]@users.noreply.github.com [in_app_pur]: Bump com.android.billingclient:billing from 6.0.0 to 6.0.1 in /packages/in_app_purchase/in_app_purchase_android/android (flutter/packages#4422)
2023-07-12 ditman@gmail.com [file_selector] Avoids using path_provider in web example app. (flutter/packages#4445)
2023-07-12 ian@hixie.ch [rfw] Add some more documentation for RFW (flutter/packages#4349)
2023-07-12 stuartmorgan@google.com [ci] Enable LUCI legacy analysis (flutter/packages#4435)
2023-07-11 ian@hixie.ch [webview_flutter_wkwebview] NSError.toString (flutter/packages#4441)
2023-07-11 stuartmorgan@google.com [ci] Remove unused Chromium setup (flutter/packages#4437)
2023-07-11 ian@hixie.ch [flutter_plugin_tools] Reimplements the excerpt system inline in the tool, rather than relying on a separate package. (flutter/packages#4417)
2023-07-11 10687576+bparrishMines@users.noreply.github.com [ci] Remove webview_flutter implementation opt outs for custom analysis (flutter/packages#4438)
2023-07-11 stuartmorgan@google.com [palette_generator] Add web support to unit tests (flutter/packages#4440)
2023-07-11 stuartmorgan@google.com [tool] Conditionalize color on `stdout` (flutter/packages#4436)
2023-07-11 47866232+chunhtai@users.noreply.github.com [go_router_builder] Cleans up builder code. (flutter/packages#4356)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com,rmistry@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 5c887028810d to 16e2ab7e986c (2 revisions) (#130421)

https://github.com/flutter/engine/compare/5c887028810d...16e2ab7e986c

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from ac4c113c071d to ebc149cff431 (4 revisions) (flutter/engine#43603)
2023-07-12 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from zTuZjXzI3bqscWmeo... to 1STsUj0X5YgpiSNEb... (flutter/engine#43602)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from zTuZjXzI3bqs to 1STsUj0X5Ygp

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

TahaTesser committed:
Update `Divider`/`VerticalDivider` and theme tests for M2/M3 (#130415)

Updated unit tests for `Divider`/ `VerticalDivider`  and theme to have M2 and M3 versions.

More info in https://github.com/flutter/flutter/pull/128725

thkim1011 committed:
Use platform specific line separator in gen-l10n (#130090)

Currently files are not generated with `\r\n` in windows. This PR should
fix the issue.

Fixes https://github.com/flutter/flutter/issues/109761.

gspencergoog committed:
Add missing links to examples that aren't linked anywhere (#130422)

## Description

This adds links to examples that were not linked anywhere.

## Related Issues
 - Fixes #129956

## Tests
 - Documentation only change

gilnobrega committed:
Reland [a11y] CupertinoSwitch On/Off labels (#130173)

This original PR (https://github.com/flutter/flutter/pull/127776) was reverted (https://github.com/flutter/flutter/pull/130166) due to a Google test failure.

This reopens the PR as per the discussion in https://github.com/flutter/flutter/pull/130166#issuecomment-1626105218.

Fixes issue #4830.

srujzs committed:
Refactor JSNumber.toDart and Object.toJS (#129436)

JSNumber.toDart will now be two functions: toDartDouble and toDartInt.

There was code that did an Object.toJS. This has been changed to
use Function.toJS as well to make it consistent with the code
in flutter/packages:
https://github.com/flutter/packages/blob/0ef393811d0c2653e68ac135733353fcad8fffa9/packages/web_benchmarks/lib/src/recorder.dart#L1223

This is to help land this CL:
https://dart-review.googlesource.com/c/sdk/+/309082

https://dart-review.googlesource.com/c/sdk/+/309081 is the CL that added
the new methods.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description
above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] All existing and new tests are passing.

QuncCccccc committed:
Update `Checkbox` tests for M2/M3 (#130351)

Updated unit tests for `Checkbox` to have M2 and M3 versions.

More info in #127064

LongCatIsLooong committed:
Exclude `Tooltip`'s overlay child from SelectableRegion (#130181)

Fixes https://github.com/flutter/flutter/issues/129969 by making tooltip text unselectable (for now). 
Also fixes some other issues uncovered when I was writing the tests.

Currently `getTransformTo` only works on ancestors. I'll try to add a new method that computes the transform from 2 arbitrary render objects in the same render tree in a follow-up PR and make `Selectable` use that method instead.

engine-flutter-autoroll committed:
Roll Flutter Engine from 16e2ab7e986c to 1b1ccdd1f527 (13 revisions) (#130458)

https://github.com/flutter/engine/compare/16e2ab7e986c...1b1ccdd1f527

2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 4e989b1564ee to bedc92598644 (1 revision) (flutter/engine#43617)
2023-07-12 flar@google.com move rtree and canvas_spy sources to Fuchsia sub-directory (flutter/engine#43615)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 68e3c0b3eea7 to 4e989b1564ee (6 revisions) (flutter/engine#43614)
2023-07-12 ian@hixie.ch Document (and assert) that channel names can't contains nulls (flutter/engine#43593)
2023-07-12 58529443+srujzs@users.noreply.github.com Reland "Refactor JSNumber.toDart and Object.toJS" (flutter/engine#43363)
2023-07-12 31859944+LongCatIsLooong@users.noreply.github.com Add a flag to `ParagraphBuilder` for rounding hack migration (flutter/engine#43118)
2023-07-12 john@johnmccutchan.com [Impeller] Fixes for asymmetric stencil descriptors (flutter/engine#43535)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from adeeb8d50f7c to 68e3c0b3eea7 (1 revision) (flutter/engine#43609)
2023-07-12 flar@google.com Add comment to use of 3x3 mapRect in TransformLayer (flutter/engine#43608)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from a251a36ea519 to adeeb8d50f7c (1 revision) (flutter/engine#43606)
2023-07-12 skia-flutter-autoroll@skia.org Roll Dart SDK from b95f6531c726 to 8f8f281ccdc6 (2 revisions) (flutter/engine#43607)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from ebc149cff431 to a251a36ea519 (2 revisions) (flutter/engine#43604)
2023-07-12 jonahwilliams@google.com [Impeller] Use new SkParagraph APIs for stroked text. (flutter/engine#41735)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

Commit:
Revert "Roll Flutter Engine from 16e2ab7e986c to 1b1ccdd1f527 (13 revisions)" (#130479)

Reverts flutter/flutter#130458

srawlins committed:
Correct a few import sortings (#130435)

Move a few import directives to be in correct order. Fixes https://github.com/flutter/flutter/issues/130434

engine-flutter-autoroll committed:
Roll Packages from 250871431bed to aa1eace00188 (6 revisions) (#130495)

https://github.com/flutter/packages/compare/250871431bed...aa1eace00188

2023-07-13 ian@hixie.ch [rfw] Restore RFW to 100% coverage (flutter/packages#4355)
2023-07-12 65381000+raju8000@users.noreply.github.com [file_selector_web] Listens to file input cancel event. (flutter/packages#3683)
2023-07-12 sam.rawlins@gmail.com [cross_file] Correct sorting of import starting with dot-slash (flutter/packages#4449)
2023-07-12 ian@hixie.ch [metrics_center] Remove Equatable dependency (flutter/packages#4444)
2023-07-12 engine-flutter-autoroll@skia.org Roll Flutter from 3ec96a8a4a5d to 544d30dbaddf (66 revisions) (flutter/packages#4448)
2023-07-12 stuartmorgan@google.com [ci] Move snippet checks to LUCI (flutter/packages#4446)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com,rmistry@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

dnfield committed:
Revert "Marks Linux firebase_oriol33_abstract_method_smoke_test to be unflaky" (#130497)

Reverts flutter/flutter#128398

See e.g.
https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20firebase_oriol33_abstract_method_smoke_test/350/overview

engine-flutter-autoroll committed:
Roll Flutter Engine from 16e2ab7e986c to b4080361f2f9 (28 revisions) (#130496)

Roll Flutter Engine from 16e2ab7e986c to b4080361f2f9 (28 revisions)

https://github.com/flutter/engine/compare/16e2ab7e986c...b4080361f2f9

2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from e5ec341bc3ca to 56b68ce6196c (1 revision) (flutter/engine#43633)
2023-07-13 skia-flutter-autoroll@skia.org Roll Dart SDK from f499e91e8cb2 to ade4dae923f3 (1 revision) (flutter/engine#43632)
2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from 811b046c673b to e5ec341bc3ca (1 revision) (flutter/engine#43631)
2023-07-13 ian@hixie.ch Make GOMA state automatic by default (flutter/engine#43584)
2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from c8da0c657c4e to 811b046c673b (3 revisions) (flutter/engine#43630)
2023-07-13 chinmaygarde@google.com [Impeller] Remove unactionable error logs and use structure chains for instance creation. (flutter/engine#43629)
2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from 7f391ea9164e to c8da0c657c4e (1 revision) (flutter/engine#43628)
2023-07-13 chinmaygarde@google.com [Impeller] Add RAII wrappers for VMA objects. (flutter/engine#43626)
2023-07-13 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from 1STsUj0X5YgpiSNEb... to xBJq6PsO5ebblODMe... (flutter/engine#43627)
2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from 6ed93436d57c to 7f391ea9164e (1 revision) (flutter/engine#43625)
2023-07-13 skia-flutter-autoroll@skia.org Roll Dart SDK from 8f8f281ccdc6 to f499e91e8cb2 (3 revisions) (flutter/engine#43623)
2023-07-13 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from 0fvk838jTDNQ_l43k... to 3C7P0w8ySmtqpyi3S... (flutter/engine#43622)
2023-07-13 skia-flutter-autoroll@skia.org Roll Skia from bedc92598644 to 6ed93436d57c (3 revisions) (flutter/engine#43621)
2023-07-13 jonahwilliams@google.com [Impeller] Add support to embedder for Impeller on GL (via Angle on Windows). (flutter/engine#43388)
2023-07-12 jonahwilliams@google.com [Impeller] Allocate buffers out of a pool on the raster thread. (flutter/engine#43564)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 4e989b1564ee to bedc92598644 (1 revision) (flutter/engine#43617)
2023-07-12 flar@google.com move rtree and canvas_spy sources to Fuchsia sub-directory (flutter/engine#43615)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from 68e3c0b3eea7 to 4e989b1564ee (6 revisions) (flutter/engine#43614)
2023-07-12 ian@hixie.ch Document (and assert) that channel names can't contains nulls (flutter/engine#43593)
2023-07-12 58529443+srujzs@users.noreply.github.com Reland "Refactor JSNumber.toDart and Object.toJS" (flutter/engine#43363)
2023-07-12 31859944+LongCatIsLooong@users.noreply.github.com Add a flag to `ParagraphBuilder` for rounding hack migration (flutter/engine#43118)
2023-07-12 john@johnmccutchan.com [Impeller] Fixes for asymmetric stencil descriptors (flutter/engine#43535)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from adeeb8d50f7c to 68e3c0b3eea7 (1 revision) (flutter/engine#43609)
2023-07-12 flar@google.com Add comment to use of 3x3 mapRect in TransformLayer (flutter/engine#43608)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from a251a36ea519 to adeeb8d50f7c (1 revision) (flutter/engine#43606)
2023-07-12 skia-flutter-autoroll@skia.org Roll Dart SDK from b95f6531c726 to 8f8f281ccdc6 (2 revisions) (flutter/engine#43607)
2023-07-12 skia-flutter-autoroll@skia.org Roll Skia from ebc149cff431 to a251a36ea519 (2 revisions) (flutter/engine#43604)
2023-07-12 jonahwilliams@google.com [Impeller] Use new SkParagraph APIs for stroked text. (flutter/engine#41735)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from 1STsUj0X5Ygp to xBJq6PsO5ebb
  fuchsia/sdk/core/mac-amd64 from 0fvk838jTDNQ to 3C7P0w8ySmtq

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md
...

Commit:
[flutter_tools] remove desktop device restrictions on Impeller. (#130430)

All current desktop backends will support Impeller after https://github.com/flutter/engine/pull/43388 lands.

vashworth committed:
Print pretty error when xcodebuild fails due to missing simulator (#130286)

Starting in Xcode 15, the simulator is no longer included in Xcode and must be downloaded and installed separately.

If you try to run flutter and the simulator is missing, you'll get an error like
```
xcodebuild: error: Unable to find a destination matching the provided destination specifier:
               		{ id:B1234A5C-67B8-901D-B2CB-FE34F56BDE78 }

               	Ineligible destinations for the "Runner" scheme:
               		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device, error:iOS 17.0 is not installed. To use with Xcode, first download and install the platform }
```

Print a pretty error to make it easier for developers to know what to do.

Part 2 of https://github.com/flutter/flutter/issues/129558.

vashworth committed:
Revert "Print pretty error when xcodebuild fails due to missing simulator" (#130504)

Reverts flutter/flutter#130286

chunhtai committed:
Unifies text field focus management in desktops (#129652)

related https://github.com/flutter/flutter/issues/128709

engine PR: https://github.com/flutter/engine/pull/43279

The web engine requires a way to unfocus textfield, It comes to nature
to me that we should leverage didGain/didLose a11y focus action. I also
unifies the action handler of all desktop platforms

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description
above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]:
https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

engine-flutter-autoroll committed:
Roll Flutter Engine from b4080361f2f9 to 88b858151685 (3 revisions) (#130502)

https://github.com/flutter/engine/compare/b4080361f2f9...88b858151685

2023-07-13 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from 3C7P0w8ySmtqpyi3S... to rRUd41Mv9NI0n3Iyc... (flutter/engine#43635)
2023-07-13 skia-flutter-autoroll@skia.org Roll Dart SDK from ade4dae923f3 to 16ddfe8d08e0 (1 revision) (flutter/engine#43634)
2023-07-13 jason-simmons@users.noreply.github.com Apply the transform of an image filter layer to paint bounds in the CanvasKit backend (flutter/engine#43353)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from 3C7P0w8ySmtq to rRUd41Mv9NI0

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

polina-c committed:
Mark leak in text_form_field_test.dart. (#130468)

LouiseHsu committed:
fix failing framework tests
